tests/e2e/support/objects/app-files/resource/actions.ts

```diff
@@ -163,6 +182,7 @@ export const editTextDocument = async ({
   await Promise.all([
     page.waitForResponse((resp) => resp.status() === 204 && resp.request().method() === 'PUT'),
     page.locator(textEditorInput).fill(content),
+    page.waitForSelector(saveTextFileInEditorButton, { state: 'visible' }),
```

Comment:
why was it necessary?
`page.locator(saveTextFileInEditorButton).click()` should wait for the selector before clicking
see https://playwright.dev/docs/api/class-locator#locator-click

Reply:
Sometime saveTextEditorButton doesn't work. This lead to intermittent failure 
```
npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/30311/tracing/upload-resources-in-personal-space-alice-2022-12-2-04-45-52.zip
```
.So to make sure the button is visible before clicking I added that state visible. As suggested by swikriti

Reply:
Then try with `'#text-editor-controls-save:visible'` here
https://github.com/owncloud/web/blob/b5a70e7cbc05bf9c3b39cab20446a4e17488c98f/tests/e2e/support/objects/app-files/resource/actions.ts#L21